HOSTEDCP-501: Added detailed documentation about Hypershift release process (#2272)
openshift/hypershift: docs/content/contribute/release-process.md

# Release process

!!! important

>     This is a complex process that involve some changes in multiple repositories and will affect multiple teams daily basis work.

Reviewer: ```suggestion
    This is a complex process that involves some changes in multiple repositories and will affect multiple teams daily basis work.
```

  - Generate release notes:
    - `GITHUB_ACCESS_TOKEN=XXXX FROM=v0.1.0 TO=v0.1.1 make release`
    - Use the output to create the PR for bump the new image in the product gitOps repo. E.g.

> This is a sample of how looks like the release notes added to the PR:

Reviewer: ```suggestion
This is a sample of how the release notes look when added to the PR:
```

  ```
  ## area/control-plane-operator

  - [cpo: cno: follow image name change in release payload](https://github.com/openshift/hypershift/pull/2230)

  ## area/hypershift-operator

  - [Added documentation around supported-versions configmap](https://github.com/openshift/hypershift/pull/2220)
  - [Add comment for BaseDomainPrefix](https://github.com/openshift/hypershift/pull/2219)
  - [Add condition to NodePool indicating whether a security group for it is available](https://github.com/openshift/hypershift/pull/2216)
  - [HOSTEDCP-827: Add root volume encryption e2e test](https://github.com/openshift/hypershift/pull/2192)
  - [fix(hypershift): reduce CAPI rbac access](https://github.com/openshift/hypershift/pull/2173)
  - [Validate Network Input for HostedCluster](https://github.com/openshift/hypershift/pull/2215)
  ```

### Supported Versions

This change involve basically the OCP supported versions by Hypershift operator:

> - `support/supportedversion/version.go` which contain the variable called `LatestSupportedVersion`. This one contains as you can imagine, the Latest supported version. We need to put the new version` here.

Reviewer: ```suggestion
- `support/supportedversion/version.go` which contains the variable called `LatestSupportedVersion`. This one contains, as you can imagine, the Latest supported version. We need to put the new version here.
```

> - `support/supportedversion/version.go` which contains the variable called `LatestSupportedVersion`. This one contains, as you can imagine, the Latest supported version. We need to put the new version` here.

Reviewer: ```suggestion
- `support/supportedversion/version.go` which contains the variable called `LatestSupportedVersion`. This one contains, as you can imagine, the Latest supported version. We need to put the new version here.
```

- `aws/create/hypershift-aws-create-chain.yaml`
- `aws/destroy-nested-management-cluster/hypershift-aws-destroy-nested-management-cluster-chain.yaml`
- `aws/destroy/hypershift-aws-destroy-chain.yaml`
- `aws/setup-nested-management-cluster/hypershift-aws-setup-nested-management-cluster-chain.yaml`
- `azure/create/hypershift-azure-create-chain.yaml`
- `azure/destroy/hypershift-azure-destroy-chain.yaml`
- `dump/hypershift-dump-chain.yaml`
- `install/hypershift-install-chain.yaml`
- `kubevirt/create/hypershift-kubevirt-create-chain.yaml`

NEED TO VERIFY THESE ONES

- `aws/run-e2e/external/hypershift-aws-run-e2e-external-chain.yaml`
- `mce/kubevirt/create/hypershift-mce-kubevirt-create-chain.yaml`
- `mce/kubevirt/destroy/hypershift-mce-kubevirt-destroy-chain.yaml`
- `mce/aws/destroy/hypershift-mce-aws-destroy-chain.yaml`
- `mce/aws/create/hypershift-mce-aws-create-chain.yaml`
- `agent/create/hostedcluster/hypershift-agent-create-hostedcluster-ref.yaml`
- `agent/destroy/hypershift-agent-destroy-chain.yaml`
- `ibmcloud/install/hypershift-ibmcloud-install-chain.yaml`

### Mirroring Images for CPO and HO

> For folder `core-services/image-mirroring/hypershift/` we need to add a mapping file for each version we want to mirror. This is a sample about how looks like this kind of [file](https://github.com/openshift/release/blob/master/core-services/image-mirroring/hypershift/mapping_hypershift_4_13). This sample points to version `4.13` and the filename is `mapping_hypershift_4_13`:

Reviewer: ```suggestion
For folder `core-services/image-mirroring/hypershift/` we need to add a mapping file for each version we want to mirror. This is a sample of how it looks like this kind of [file](https://github.com/openshift/release/blob/master/core-services/image-mirroring/hypershift/mapping_hypershift_4_13). This sample points to version `4.13` and the filename is `mapping_hypershift_4_13`:
```

```
registry.ci.openshift.org/ocp/4.13:hypershift quay.io/hypershift/hypershift:4.13
registry.ci.openshift.org/ocp/4.13:hypershift-operator quay.io/hypershift/hypershift-operator:4.13
registry.ci.openshift.org/ocp-arm64/4.13:hypershift quay.io/hypershift/hypershift:4.13-arm64 quay.io/hypershift-arm64/hypershift:latest-arm64
registry.ci.openshift.org/ocp-arm64/4.13:hypershift-operator quay.io/hypershift/hypershift-operator:4.13-arm64
```

As you can see, the version is hardcoded but, what happens to the `latest` version?:

```
registry.ci.openshift.org/ocp/4.14:hypershift quay.io/hypershift/hypershift:4.14 quay.io/hypershift/hypershift:latest
registry.ci.openshift.org/ocp/4.14:hypershift-operator quay.io/hypershift/hypershift-operator:4.14  quay.io/hypershift/hypershift-operator:latest

> registry.ci.openshift.org/ocp-arm64/4.14:hypershift quay.io/hypershift/hypershift:4.14-arm64 quay.io/hypershift-arm64/hypershift:latest-arm64

Reviewer: ```suggestion
registry.ci.openshift.org/ocp-arm64/4.14:hypershift quay.io/hypershift/hypershift:4.14-arm64 quay.io/hypershift/hypershift:latest-arm64
```
That repo was a typo and should be resolved once https://github.com/openshift/release/pull/36933 is merged.

Author: Cool, nice catch
